pyxrf/model/setting.py

```diff
@@ -181,18 +313,10 @@ def update_roi(self, element_list, std_ratio=4):
             self.roi_dict.update({v: roi})
 
         # remove old items not included in element_list
-        for k in six.iterkeys(self.roi_dict):
+        for k in self.roi_dict.copy().keys():
```

Comment:
Nice catch!

Reply:
This is one of many things in the original code that did not work as expected.